models/curvefi/optimism/curvefi_optimism_pools.sql

```diff
@@ -0,0 +1,127 @@
+{{ config(
+    alias = 'pools',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['version', 'tokenid', 'token', 'pool'],
```

Comment:
@MSilb7 looks like incremental logic is complaining about the ```unique_key``` being all four fields in the table -- seems it has nothing to insert/update based on that. can we reduce fields in ```unique_key``` or change this from incremental to another format?

Reply:
Hey, sorry just getting back to this. The updates to this table should be fairly rare (just when new pools are created).

Do you have a suggestion for another format type?

Reply:
incremental is fine, you just need to reduce the keys in unique keys. all columns can't be included it appears

Reply:
cool, removed 'version' from the key.

Reply:
it seems ```version``` should actually be part of the key since it's hardcoded and never changes -- are there no other columns to remove from key? you mentioned it only changes when new pools are created. whichever column indicates that changing pool, that should be the non-key value(s)

think of the unique keys as the columns to include in join conditions when writing an update statement. if this is an incremental model, the question is:
- which field(s) used in join conditions to itself? those are unique keys
- which field(s) get updated? those are non-keys

Reply:
@MSilb7 this comment here

Reply:
ah got it. I wouldn't expect any of these fields to change. So would it make sense to not do a "merge" strategy then?

Reply:
yeah, i'd revise and look at other options such as append only or insert overwrite: https://docs.getdbt.com/docs/build/incremental-models#about-incremental_strategy